packages/core/src/new-api/types/artifact.ts

```diff
@@ -19,6 +19,126 @@ export interface Artifact {
  * @beta
  */
 export interface ArtifactResolver {
-  load(contractName: string): Promise<Artifact>;
-  resolvePath(contractName: string): Promise<string>;
+  loadArtifact(contractName: string): Promise<Artifact>;
+  getBuildInfo(contractName: string): Promise<BuildInfo | undefined>;
+  resolvePath(contractName: string): Promise<string | undefined>;
```

Comment:
Already mentioned this in a previous PR, but ideally we shouldn't use these paths. Not using them adds flexibility to the implementer of this interface (i.e. integrators of ignition)